pkg/api/defaults-kubelet.go

```diff
@@ -193,6 +205,18 @@ func (cs *ContainerService) setKubeletConfig(isUpgrade bool) {
 			}
 		}
 
+		if isUpgrade && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.14.0") {
+			hasSupportPodPidsLimitFeatureGate := strings.Contains(o.KubernetesConfig.KubeletConfig["--feature-gates"], "SupportPodPidsLimit=true")
```

Comment:
This should get the --feature-gates flag from the agent profile

Reply:
Same as above.